src/cockpit/channels/filesystem.py

```diff
@@ -378,6 +378,20 @@ def get_group(gid: int) -> 'str | int':
             except KeyError:
                 return gid
 
+        def get_access(name: str, fd: int, mode: int, *, follow_symlinks: bool = False) -> 'bool | None':
+            if not name:
+                # HACK: Python's os.access() does not support passing "AT_EMPTY_PATH"
+                # so we need to resolve the name of the watched directory.
+                try:
+                    name = os.readlink(f'/proc/self/fd/{fd}')
```

Comment:
Any reason you can't just `os.access()` directly on the `/proc/self/fd` link with `follow_symlinks=True`?  That underlying file descriptor is exactly what you want to query.  Trying to do it via filename is possibly racy and I imagine it could run into corner cases...